src/parser/lval.js

```diff
@@ -84,6 +84,8 @@ export default class LValParser extends NodeUtils {
           break;
 
         case "SpreadElement":
+          this.checkToRestConversion(node);
```

Comment:
this is supposed to fix https://github.com/babel/babylon/issues/667 , it **probably** could be checked inside of the `toAssignable` call 3 lines below, but as mentioned in the other comment - im not sure whats the best way to share parent's (in this example parent being rest element) context/state information